### PR TITLE
Adding transfer effects so creators can transfer in/out by slot.

### PR DESCRIPTION
**Pull request by blockiosaurus**

### CODE REVIEW

**Comment by 0xLT** [comment]
Looks good, very simple, very effective.

You seem to be missing the change to transfer out, though.

**Comment by blockiosaurus** [comment]
> Looks good, very simple, very effective.
> 
> You seem to be missing the change to transfer out, though.

So I am, haha

File: program/src/state/transfer_effects.rs [hunk 1]
```diff
@@ -7,7 +7,9 @@ pub struct TransferEffects {
     pub burn: bool,
     pub freeze: bool,
     pub freeze_parent: bool,
-    pub empty_bytes: B12,
+    pub auth_transfer_in: bool,
+    pub auth_transfer_out: bool,
+    pub empty_bytes: B10,
```
Comment (0xLT) [inline program/src/state/transfer_effects.rs:12]: why is this a change from b12 to b10?

**Review by 0xLT — CHANGES_REQUESTED** [review]
still waiting for the `processor/transfer_out` change

File: program/src/state/transfer_effects.rs (hunk 1 above)
Comment (blockiosaurus) [reply program/src/state/transfer_effects.rs:12]: We're using a bitfield for transfer effects to save space. So we have allocated 16 bits of space to the field, which we need to declare in the type. So before we said we have 4 flags and 12 bits of empty space. Now we're saying we have 6 flags and 10 bits of empty space. Those aren't actually bools, they're just single bits.

Comment (0xLT) [reply program/src/state/transfer_effects.rs:12]: ah i see. ok, looks good to me.

**Review by 0xLT — APPROVED** [review]

**Review by febo — APPROVED** [review]
LGTM

**Pull request merged**